src/main/java/org/openstreetmap/atlas/geography/atlas/builder/proto/ProtoAtlasBuilder.java

```diff
@@ -0,0 +1,232 @@
+import org.openstreetmap.atlas.proto.ProtoLine;
+import org.openstreetmap.atlas.proto.ProtoLocation;
+import org.openstreetmap.atlas.proto.ProtoPoint;
+import org.openstreetmap.atlas.streaming.resource.Resource;
+import org.openstreetmap.atlas.streaming.resource.WritableResource;
+import org.openstreetmap.atlas.utilities.collections.Maps;
+import org.openstreetmap.atlas.utilities.conversion.ProtoLocationConverter;
+import org.openstreetmap.atlas.utilities.conversion.ProtoTagListConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+
+/**
+ * Build an Atlas from a naive proto serialized file, or write an Atlas to a naive proto serialized
+ * file. At this stage, the builder can only handle Atlases that have tagged Points and Lines.
+ *
+ * @author lcram
+ */
+public class ProtoAtlasBuilder
+{
+    private static final Logger logger = LoggerFactory.getLogger(ProtoAtlasBuilder.class);
+
+    /**
+     * Read a resource in naive ProtoAtlas format into a PackedAtlas.
+     *
+     * @param resource
+     *            the resource in naive ProtoAtlas format
+     * @return the constructed PackedAtlas
+     */
+    public PackedAtlas read(final Resource resource)
+    {
+        ProtoAtlasContainer protoAtlasContainer = null;
+
+        // First, we need to construct the container object from the proto binary
+        try
+        {
+            protoAtlasContainer = ProtoAtlasContainer.parseFrom(resource.readBytesAndClose());
+        }
+        catch (final InvalidProtocolBufferException exception)
+        {
+            throw new CoreException("Error deserializing the ProtoAtlasContainer from {}",
+                    resource.getName(), exception);
+        }
+
+        // initialize atlas metadata
+        final long numberOfPoints = protoAtlasContainer.getNumberOfPoints();
+        final long numberOfLines = protoAtlasContainer.getNumberOfLines();
+        final long numberOfAreas = protoAtlasContainer.getNumberOfAreas();
+        final long numberOfNodes = 0;
+        final long numberOfEdges = 0;
+        final long numberOfRelations = 0;
+        final AtlasSize atlasSize = new AtlasSize(numberOfEdges, numberOfNodes, numberOfAreas,
+                numberOfLines, numberOfPoints, numberOfRelations);
+        // TODO it would be nice to programmatically determine which protobuf version is in use
+        final AtlasMetaData atlasMetaData = new AtlasMetaData(atlasSize, true, "unknown",
+                "ProtoAtlas", "unknown", "unknown", Maps.hashMap());
+        final PackedAtlasBuilder builder = new PackedAtlasBuilder().withSizeEstimates(atlasSize)
+                .withMetaData(atlasMetaData).withName(resource.getName());
+
+        // build the atlas features
+        parsePoints(builder, protoAtlasContainer.getPointsList());
+        parseLines(builder, protoAtlasContainer.getLinesList());
+        parseAreas(builder, protoAtlasContainer.getAreasList());
+
```

Comment:
Are there no relations being added to the new packed Atlas or am I missing something?

Reply:
We're getting there. It's still in development :)